syntax = "proto3";
package common;

// Shared message for raw energy data
message EnergyRecord {
  string timestamp = 1;
  string household_id = 2;
  double power_consumption = 3;
  double voltage = 4;
  double current = 5;
}

// Shared message for analyzed energy data
message AnalyzedRecord {
  EnergyRecord original = 1;
  double efficiency_score = 2;
  bool is_anomaly = 3;
  string anomaly_reason = 4;
}